cpp/src/io/orc/dict_enc.cu

```diff
@@ -127,10 +127,8 @@ __global__ void __launch_bounds__(block_size, 2)
   uint32_t nnz, start_row, dict_char_count;
   int t = threadIdx.x;
 
-  if (t < sizeof(DictionaryChunk) / sizeof(uint32_t)) {
-    ((volatile uint32_t *)&s->chunk)[t] =
-      ((const uint32_t *)&chunks[group_id * num_columns + col_id])[t];
-  }
+  if (t == 0) s->chunk = chunks[group_id * num_columns + col_id];
+  // TO-DO, if possible have to simplify
```

Comment:
Not really. This is a hash map being initialized. It's different from cooperative load/store and is scope creep. It will probably be refactored out in a separate PR when we abstract away hash maps.